src/main/daml/Daml/Finance/Instrument/Generic/Election.daml

```diff
@@ -69,64 +66,39 @@ template Election
         description
         eventTime = electionTime
 
--- | Helper contract to delegate the right to create `Election`\s referencing a specific
--- `Instrument`. The provider delegates the ability to create `Election`\s to any party that has
--- visibility on the `ElectionFactory` contract. In order to create the `Election`, a valid
--- `Holding` must be presented which identifies the choice controller as either the owner or the
--- custodian to the `Holding`.
-template ElectionFactory
+-- | Factory template to create an `Election`.
+template Factory
   with
     provider : Party
-      -- ^ The provider.
-    id : Id
-      -- ^ Election identifier
-    description : Text
-      -- ^ A human readable description of the election claim.
-    claim : C
-      -- ^ the elected sub-tree
-    observers : Parties
-      -- ^ observers of the contract
-    instrument : InstrumentKey
-      -- ^ key of the instrument to which the election applies
+      -- ^ The provider of the `Factory`.
+    observers : PartiesMap
+      -- ^ A set of observers.
   where
     signatory provider
-    observer observers
+    observer Disclosure.flattenObservers observers
+
+    interface instance Factory.F for Factory where
+      view = Factory.View with provider
+      create' arg =
```

Comment:
We typically expand the `arg` to get direct access to each argument instead of using dot notation. Although I don't think this is in the style guide...

Reply:
Perhaps it's better to keep it consistent as this would be the only Factory implementation not to follow that pattern?